### PR TITLE
plan: first step of removing parents

### PR DESCRIPTION
**Pull request by hanfei1991**
Parents seem to be redundant in planner, which is only used in ppd and explain. Removing it can help to clean code and increase efficiency.

### CODE REVIEW

File: plan/predicate_push_down.go [hunk 1]
```diff
@@ -44,10 +46,8 @@ func (p *baseLogicalPlan) PredicatePushDown(predicates []expression.Expression)
 		return predicates, p.self
 	}
 	child := p.children[0].(LogicalPlan)
-	rest, _ := child.PredicatePushDown(predicates)
-	if len(rest) > 0 {
-		addSelection(p.self, child, rest)
-	}
+	rest, ch := child.PredicatePushDown(predicates)
```
Comment (coocood) [inline plan/predicate_push_down.go]: `ch` is used frequently as `channel`.
How about `newChild`?

File: plan/predicate_push_down.go [hunk 2]
```diff
@@ -44,10 +46,8 @@ func (p *baseLogicalPlan) PredicatePushDown(predicates []expression.Expression)
...
+	addSelection(p.self, ch, rest, 0)
```
Comment (shenli) [inline plan/predicate_push_down.go]: Why not check the length of rest?

Comment (hanfei1991) [reply plan/predicate_push_down.go]: chek it inside function

File: plan/predicate_push_down.go [hunk 3]
```diff
@@ -377,8 +355,8 @@ func (p *LogicalAggregation) PredicatePushDown(predicates []expression.Expressio
 			ret = append(ret, cond)
 		}
 	}
-	_, _ = p.baseLogicalPlan.PredicatePushDown(condsToPush)
-	return ret, retPlan
+	p.baseLogicalPlan.PredicatePushDown(condsToPush)
```
Comment (shenli) [inline plan/predicate_push_down.go]: The return value is ignored.

**Comment by coocood** [comment]
LGTM

File: plan/predicate_push_down.go (hunk 3 above)
Comment (shenli) [reply plan/predicate_push_down.go]: It is handled by its base class.

**Comment by shenli** [comment]
LGTM

**Pull request merged**